fs/layer/layer.go

```diff
@@ -327,6 +327,7 @@ func (r *Resolver) Resolve(ctx context.Context, hosts []docker.RegistryHost, ref
 	if err != nil {
 		return nil, err
 	}
+	ztoc.TOC.FileMetadata = nil // We no longer need to hold onto the TOC
```

Comment:
Done, I would prefer this gets passed into the metadataStore function, but since it's a function that must be implemented by whatever store uses it, I figured it was better to just consume it after using it.